build_ext/lint.py

```diff
@@ -0,0 +1,385 @@
+# Copyright (c) 2016 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+import ast
+import re
+
+from distutils.spawn import spawn
+from distutils.text_file import TextFile
+
+from build_ext.utils import Utils, BaseCommand, memoize
+
+try:
+    # These dependencies aren't available in build environments.  We won't need any
+    # linting functionality there though, so just create a dummy class so we can proceed.
+    from pep8 import Checker, register_check
```

Comment:
`register_check` should be okay because that's only invoked from within the `PluginLoadingFlake8Command.run` method which never gets called during a build.  During a build we should hit the import error, create the dummy classes, and then just keep going.

The overall problem was that when we imported the `lint` module from `setup.py`, Python would then try to import everything `lint` was importing (much of which was unavailable).  Normally we could solve this by dropping the import statements down into a method, but that wouldn't work in the case where we had classes in `lint` that were descended from classes in `flake8` and `pep8`.  So we create these dummy classes on an import error so that everything resolves but of course none of it would actually work if it were run in that state.